### PR TITLE
Clear realpath cache when reading the token from file

### LINKED ISSUE #900: Intermittent failure while loading token in EKS with service accounts
I'm running a Symfony app in EKS, and I'm using SQS with Messenger + this lib.

I'm having small clusters of failures at long intervals with 403 responses from the SQS APIs:
```
HTTP 403 returned for "https://sqs.eu-central-1.amazonaws.com/".

Code:    AccessDenied
Message: Access to the resource https://sqs.eu-central-1.amazonaws.com/ is denied.
Type:    Sender
Detail:
```
I've dig into the logs, and I've found this warning:
```
php.WARNING: Warning: file_get_contents(/var/run/secrets/eks.amazonaws.com/serviceaccount/token): failed to open stream: No such file or directory {"exception":"[object] (ErrorException(code: 0): Warning: file_get_contents(/var/run/secrets/eks.amazonaws.com/serviceaccount/token): failed to open stream: No such file or directory at /var/www/my-project/vendor/async-aws/core/src/Credentials/WebIdentityProvider.php:92)"} []
```
So I've traced down the issue to something that happened on the official SDK too: https://github.com/aws/aws-sdk-php/issues/2014

Basically, it's a combination of:
 * tokens are short lived
 * EKS replaces tokens on-the-fly
 * PHP's realpath cache gets in the way...

... so you get brief intervals in which you try to read the old, now missing token.

In the offical SDK, the issue was mitigated by clearing the cache in a precise and ordered way of the following paths:
```
/var/run/secrets/eks.amazonaws.com/serviceaccount/..data/token
/var/run/secrets/eks.amazonaws.com/serviceaccount/..data
/var/run/secrets/eks.amazonaws.com/serviceaccount/token
```
in this PR: https://github.com/aws/aws-sdk-php/pull/2043

Can we apply the same logic here too?

### PR DESCRIPTION
**Pull request by Jean85**
This fixes #900 by applying the same fix of the official SDK in aws/aws-sdk-php#2043

### CODE REVIEW

**Review by jderusse — APPROVED** [review]
thank you for finding the reason of this issue.

I'd use an intermediate variable `$tokenDir` and `tokenLink` to avoid calling the methods twice. 
But this is just a nitpicking.

**Comment by Nyholm** [comment]
Thank you for the issue and the PR. Well done.

**Comment by Jean85** [comment]
Thank you! Variables added.

**Review by jderusse — APPROVED** [review]

File: src/Core/src/Credentials/WebIdentityProvider.php [hunk 1]
```diff
@@ -122,4 +120,26 @@ private function getCredentialsFromRole(string $roleArn, string $tokenFile, ?str
...
+        if (!file_exists($tokenFile)) {
+            $tokenDir = \dirname($tokenFile);
+            $tokenLink = readlink($tokenFile);
+            clearstatcache(true, $tokenDir . \DIRECTORY_SEPARATOR . $tokenLink);
+            clearstatcache(true, $tokenDir . \DIRECTORY_SEPARATOR . \dirname($tokenLink));
+            clearstatcache(true, $tokenFile);
+        }
+
+        if (false === $token = file_get_contents($tokenFile)) {
+            throw new RuntimeException('failed to read data');
```
Comment (Nyholm) [inline src/Core/src/Credentials/WebIdentityProvider.php:140]:
```suggestion
            throw new RuntimeException('Failed to read data');
```

**Review by Nyholm — APPROVED** [review]
Thank you. 

Is this a bug fix or a feature?

**Comment by Jean85** [comment]
@Nyholm I would consider this a bugfix.

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**